pkg/messages/treatyMessage.go

```diff
@@ -0,0 +1,116 @@
+package messages
+
+import (
+	"github.com/google/uuid"
+)
+
+type ConditionType int // type of condition
+type RequestType int
+type Op int
+
+const (
+	HP ConditionType = iota + 1
+	Floor
+	AvailableFood
+)
+
+const (
+	LeaveAmountFood RequestType = iota + 1
+	LeavePercentFood
+	Inform
+)
+
+const (
+	GT Op = iota + 1
+	GE
+	EQ
+	LE
+	LT
+)
+
+type Treaty struct {
```

Comment:
Said in the comments (and I could be wrong on this) but I think this should be in utils and imported, as agents will have to import this data type. Ik we import messages but I'm annoying

Reply:
Ehh personally I don't see the point, is there a use case you can think of for why this is better?

Reply:
Mainly convention, since we have healthInfo as a struct in a separate file and also bc I'd rather use treaties.Treaty instead of messages.Treaty. Shouldn't be mixing the 2 up

Reply:
I thought we had agreed that a treaty was a substitute of messages since you can send a receive a treaty? (it's basically a message on steroids)

If that's the case, I think it's best to keep it as is in the messages package

Reply:
I just think treaties are used outside of messages too much to be kept in here